template.yaml

```diff
@@ -48,4 +48,76 @@ Resources:
         - Key: !Sub "tag:${TagName}"
           Values:
             - !Ref TagValue
-      AutomationTargetParameterName: InstanceId
\ No newline at end of file
+      AutomationTargetParameterName: InstanceId
+
+  RequireDeleteTag:
+    Type: AWS::Config::ConfigRule
+    Properties:
+      ConfigRuleName: RequireDeleteTag
+      Description: !Sub "Checks whether the '${TagName}' tag is set for EC2 Instances."
+      Source:
+        Owner: AWS
+        SourceIdentifier: REQUIRED_TAGS
+      Scope:
+        ComplianceResourceTypes:
+          - AWS::EC2::Instance
+      InputParameters:
+        tag1Key: !Ref TagName
+
+  TagUntagedInstances:
+    Type: "AWS::Config::RemediationConfiguration"
+    Properties:
+      Automatic: true
+      ConfigRuleName: !Ref RequireDeleteTag
+      MaximumAutomaticAttempts: 5
+      RetryAttemptSeconds: 50
+      Parameters:
+        AutomationAssumeRole:
+          StaticValue:
+            Values:
+              - !GetAtt AutomationServiceRole.Arn
+        InstanceId:
+          ResourceValue:
+            Value: RESOURCE_ID
+      TargetId: !Ref TagInstancesDocument
+      TargetType: "SSM_DOCUMENT"
+
+  AutomationServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ssm.amazonaws.com
+                - ec2.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AmazonSSMAutomationRole
+      Path: "/"
+      RoleName: AutomationServiceRole
+
+  TagInstancesDocument:
+    Type: AWS::SSM::Document
+    Properties:
+      DocumentType: Automation
+      Content:
+        schemaVersion: '0.3'
+        assumeRole: "{{ AutomationAssumeRole }}"
+        parameters:
+          InstanceId:
+            type: String
+          AutomationAssumeRole:
+            type: String
+        mainSteps:
+          - name: TagEC2Instance
+            action: aws:createTags
+            inputs:
+              ResourceType: EC2
+              ResourceIds:
+                - "{{ InstanceId }}"
+              Tags:
+                - Key: !Ref TagName
+                  Value: !Ref TagValue
```

Comment:
Gibt es ne moeglichkeit in CF, einen parameter zu einem anderen zu defaulten?
An dieser Stelle wuerde denke ich, koennte man einen optionalen parameter der Art `TagDefault` einfuehren, der hier genutzt wird wenn gesetzt und sonst `TagValue`

Reply:
Ich meinte das so, dass die nicht getaggten Ressourcen im Standardfall immer abends gestoppt werden. Nur wenn man den Tag explizit auf etwas anderes gesetzt hat, dann bleibt die Instanz erhalten. 
Dafür muss es dann ja schon der selbe `TagValue` sein, wie hier: https://github.com/superluminar-io/cf-ssm-example/blob/49165318787e7f78979addf94e19aa08283c282c/template.yaml#L50

Reply:
Stimmt, wenn das das ziel ist, voellig richtig :D